core-rust-bridge/src/main/java/com/radixdlt/transaction/REv2TransactionAndProofStore.java

```diff
@@ -100,9 +100,11 @@ public REv2TransactionAndProofStore(Metrics metrics, NodeRustEnvironment nodeRus
             .build(new TypeToken<>() {});
   }
 
-  public Option<TxnsAndProof> getTxnsAndProof(long startStateVersionInclusive) {
+  public Option<TxnsAndProof> getTxnsAndProof(
```

Comment:
This is fine, although minorly surprising - I'd kinda expect this config to live in some kind of rust service responsible for ledger sync (and set from some kind of constructor).

I don't think we should change that in this PR necessarily, but just a thought for the future 👍

Reply:
> to live in some kind of rust service responsible for ledger sync

Yeah, that's the problem - there is no Rust service responsible for ledger sync 🤷 (because Java is).

This PR merely wraps the 2 ints into a `LedgerSyncLimitsConfig`, but I agree that some coming-soon refactor could introduce a... "Rust service responsible for _picking transactions to be returned in_ ledger sync" - if it's not too narrow of a responsibility :(
A much better move will be possible only after migrating to 100% Rust.